assign-doi-datacite/src/main/java/no/unit/nva/datacite/mdsclient/DataCiteMdsConnection.java

```diff
@@ -0,0 +1,218 @@
+ *
+ * <p>The HttpClient provided should have a {@link java.net.Authenticator} associated to do pre-emptive
+ * authentication towards the API server.
+ *
+ * <p>Use the {@link DataciteMdsConnectionFactory#getAuthenticatedConnection(String)} to construct new instances.
+ */
+public class DataCiteMdsConnection {
+
+    public static final String HTTPS = "https";
+    public static final String DATACITE_PATH_DOI = "doi";
+    public static final String DATACITE_PATH_METADATA = "metadata";
+    public static final String FORM_PARAM_DOI = "doi";
+    public static final String FORM_PARAM_URL = "url";
+
+    public static final String CHARACTER_SLASH = "/";
+    public static final String APPLICATION_XML_CHARSET_UTF_8 = "application/xml; charset=UTF-8";
+
+    private final transient HttpClient httpClient;
+    private final String host;
+    private final int port;
+
+    /**
+     * Constructor for testability reasons.
+     *
+     * @param httpClient HttpClient
+     */
+    public DataCiteMdsConnection(HttpClient httpClient, String host, int port) {
+        this.httpClient = httpClient;
+        this.host = host;
+        this.port = port;
+    }
+
+    /**
+     * This request stores a new version of metadata.
+     *
+     * @param doi         prefix/suffix
+     * @param dataciteXml resource metadata as Datacite XML, encoded with UTF-8.
+     * @return HttpResponse
+     * @throws IOException          IOException
+     * @throws URISyntaxException   URISyntaxException
+     * @throws InterruptedException InterruptedException
+     */
+    public HttpResponse<String> postMetadata(String doi, String dataciteXml) throws IOException,
+                                                                                    URISyntaxException,
+                                                                                    InterruptedException {
+
+        URI uri = new URIBuilder()
+            .setScheme(HTTPS)
+            .setHost(host)
+            .setPort(port)
+            .setPath(DATACITE_PATH_METADATA + CHARACTER_SLASH + doi)
+            .build();
+
+        HttpRequest request = HttpRequest.newBuilder()
+            .POST(HttpRequest.BodyPublishers.ofString(dataciteXml))
+            .uri(uri)
+            .header(HttpHeaders.CONTENT_TYPE, APPLICATION_XML_CHARSET_UTF_8)
+            .build();
+
+        return httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+    }
+
+    /**
+     * This request requests the most recent version of metadata associated with a given DOI.
+     *
+     * @param doi prefix/suffix
+     * @return CloseableHttpResponse
+     * @throws IOException          IOException
+     * @throws URISyntaxException   URISyntaxException
+     * @throws InterruptedException InterruptedException
+     */
+    public HttpResponse<String> getMetadata(String doi) throws IOException, URISyntaxException, InterruptedException {
+        URI uri = new URIBuilder()
```

Comment:
extract method

Reply:
Refactored with extracting to a few helper methods. PTAL.